matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/actions/MegolmSessionDataImporter.kt

```diff
@@ -46,19 +45,14 @@ internal class MegolmSessionDataImporter @Inject constructor(private val olmDevi
     @WorkerThread
     fun handle(megolmSessionsData: List<MegolmSessionData>,
                fromBackup: Boolean,
-               uiHandler: Handler,
                progressListener: ProgressListener?): ImportRoomKeysResult {
         val t0 = System.currentTimeMillis()
 
         val totalNumbersOfKeys = megolmSessionsData.size
         var lastProgress = 0
         var totalNumbersOfImportedKeys = 0
 
-        if (progressListener != null) {
-            uiHandler.post {
-                progressListener.onProgress(0, 100)
-            }
-        }
+        progressListener?.onProgress(0, 100)
```

Comment:
Could you please add in the Javadoc of [ProgressListener](https://github.com/vector-im/riotX-android/blob/15dc4d6369834b3f1a219152a41422685c97f65f/matrix-sdk-android/src/main/java/im/vector/matrix/android/api/listeners/ProgressListener.kt#L27) that the method will not be called on the UI thread.

Also it can be worth to add the info in the file CHANGES.md, in the section `SDK API changes`